code/AEPSDK.brs

```diff
@@ -247,6 +247,48 @@ function AdobeAEPSDKInit(taskNode = invalid as dynamic) as object
             m._private.dispatchEvent(event)
         end function,
 
+        ' *************************************************************************************
+        '
+        ' set consent.
+        '
+        ' This function will send the consent data in XDM format following Adobe 2.0 standard to the Edge network.
+        '
+        '
+        ' @param data as object : containing consent XDM data following Adobe 2.0 standard
+        ' @param [optional] callback as function(context, result) : handle Edge consent response
+        ' @param [optional] context as dynamic : context to be passed to the callback function
+        '
+        ' *************************************************************************************
+
+        setConsent: function(data as object, callback = _adb_defaultCallback as function, context = invalid as dynamic) as void
+            _adb_logDebug("API: setConsent()")
+
+            if _adb_isEmptyOrInvalidMap(data) then
+                _adb_logError("setConsent() - Invalid event data, must be an associative array")
+                return
+            end if
+
+            if _adb_isEmptyOrInvalidArray(data.consent) then
+                _adb_logError("sendEvent() - Invalid event data, consent data is required.")
+                return
+            end if
+
+            ' event data: { "consent": [ { "standard": "Adobe", "version": "2.0", "value": { "collect": { "val": "y" }, "metadata": { "time": "YYYY-03-17T15:48:42-07:00" } } } ] }
+            event = _adb_RequestEvent(m._private.cons.PUBLIC_API.SET_CONSENT, data)
```

Comment:
What format are users expected to set their Consent preferences? In iOS and Android we use a custom format of `consents.collect.val` for example. For Roku are you planning on using the same format or the actual format which is sent to the Edge Network (e.g. `consent.value.collect.val`)? 
Also, will the SDK automatically set the `standard`, `version`, and `metadata.time` or are users expected to pass those in as well?

Reply:
Planning to use the actual format sent to Edge Network. Regarding the standard we decided we would only allow 2.0. But I will still work with Mitch to decide if we want to auto populate standard, version and metadata.time.

If there is a slightest possibility we would accept other consent standards, we should get the entire payload from customer. I will inform you on what Mitch says. In the meantime do let me know what do you think about this as well.

Reply:
One issue I see is the request could get rejected if the customer sets the timestamp in the wrong format. Unless BrightScript has an easy way to format the timestamp to what we expect, we may need to either document it well or provide a helper to get the timestamp in the correct format so it may be passed in.

Reply:
Good point, we will make sure to have this documented properly.